File: tournaments/increaseNumberRoundness/increaseNumberRoundness.js
function increaseNumberRoundness(n) {
    let a = 0;
    let b = 0;

    while (0 < n) {
        let c = n % 10;
        n = ~~(n / 10);

        if (!c) {
            if (b) a = true;
        } else {
            if (a) return true;
            else b = true;
        }
    }

    return false;
}

File: tournaments/increaseNumberRoundness/increaseNumberRoundness.cs
bool increaseNumberRoundness(int n) {
    var isZeroFound = false;
    var isNotZeroFound = false;

    while (0 < n) {
        var nextDigit = n % 10;
        n /= 10;

        if (0 == nextDigit) {
            if (isNotZeroFound) {
                isZeroFound = true;
            }
        } else {
            if (isZeroFound) {
                return true;
            } else {
                isNotZeroFound = true;
            }
        }
    }

    return false;
}
